packages/hooks/src/useWalletsList.ts

```diff
@@ -17,11 +20,36 @@ const useWalletList = () => {
 
         // Modify the wallets to include the missing balance from the API response
         // Should remove this once the API is fixed
-        const modified_wallets = wallets?.map(wallet => ({
-            ...wallet,
-            balance: 1000,
-            landing_company_shortcode: wallet.landing_company_name,
-        }));
+        const modified_wallets =
+            wallets?.map(wallet => {
+                const currency = wallet?.currency || 'USD';
+                const is_crypto_currency = is_crypto(currency);
+                const is_fiat = !is_crypto_currency;
+                const icon = getWalletCurrencyIcon(wallet.is_virtual ? 'demo' : currency, is_dark_mode_on);
+                const modal_icon = getWalletCurrencyIcon(wallet.is_virtual ? 'demo' : currency, is_dark_mode_on, true);
+                const name = `${wallet.is_virtual ? 'Demo ' : ''}${currency} ${'Wallet'}`;
+
+                return {
+                    ...wallet,
+                    currency,
+                    /** Indicating whether the wallet is the currently selected wallet. */
+                    is_selected: wallet.loginid === loginid,
+                    /** Wallet balance */
+                    balance: balance_data?.balance?.accounts?.[wallet.loginid || '']?.balance || 0,
+                    /** Landing company shortcode the account belongs to. */
+                    landing_company_name:
+                        wallet.landing_company_name === 'maltainvest' ? 'malta' : wallet.landing_company_name,
+                    is_disabled: Boolean(wallet.is_disabled),
+                    is_virtual: Boolean(wallet.is_virtual),
+                    is_crypto: is_crypto_currency,
+                    is_fiat,
```

Comment:
Do we need both `is_crypto` and `is_fiat`? One is just negation of the other right?

Reply:
@nijil-deriv remove it since we don't use it in this PR